src/controllers/departments.controller.js

```diff
@@ -0,0 +1,174 @@
+import {
+  changeRecordData,
+  createRecord,
+  deleteRecord,
+  getUserBySession,
+  increaseValueBy,
+  hasDublicatedValue,
+  queryRecords,
+} from '../helpers/index.js';
+import { dbTablesNamesEnum, rolesEnum } from '../shared/index.js';
+import {
+  defaultDepartmentScheme,
+  queryDepartmentScheme,
+} from '../schemas/index.js';
+import { newActualManagerValidator } from '../validators/index.js';
+import { departmentResourceFieldsNames } from './shared/index.js';
+import { getSchemeFields } from './helpers.js';
+
+export async function createDepartmentController(req, res, next) {
+  try {
+    const scheme = getSchemeFields(req, defaultDepartmentScheme);
+    const hasActualManagerIdField = Object.hasOwn(
+      scheme.body,
+      'actualManagerID'
+    );
+    const managerID = scheme.body.actualManagerID;
+
+    if (hasActualManagerIdField) {
+      if (!(await newActualManagerValidator(managerID, res))) return;
+    }
+
+    if (await isContactNumberDuplicated(scheme.body.contactNumber)) {
+      res.statusMessage = 'Phone number already has been used';
+      return res.status(409).send();
+    }
+
+    const id = await createRecord(dbTablesNamesEnum.DEPARTMENTS, scheme.body);
+
+    if (hasActualManagerIdField) {
+      await setDepartmentIdOfManager(managerID, id);
+      await increaseEmployeeAmountByOne(id);
+    }
+
+    res.status(201).send({ id });
+  } catch (e) {
+    next(e);
+  }
+}
+
+export async function queryDepartmentsController(req, res, next) {
+  try {
+    const scheme = getSchemeFields(req, queryDepartmentScheme);
+    const valuesToQuery = { ...scheme.query };
+    delete valuesToQuery.pageSize;
+    delete valuesToQuery.pageNumber;
+
+    const results = await queryRecords(
+      dbTablesNamesEnum.DEPARTMENTS,
+      valuesToQuery,
+      departmentResourceFieldsNames
+    );
+
+    if (
+      Object.hasOwn(scheme.query, 'pageSize') &&
+      Object.hasOwn(scheme.query, 'pageNumber')
+    ) {
+      const pageSize = scheme.query.pageSize;
+      const pageNumber = scheme.query.pageNumber;
+
+      return res.status(200).send({
+        allResultsAmount: results.length,
+        results: results.slice(
+          pageSize * pageNumber,
+          pageSize * (pageNumber + 1)
```

Comment:
Not sure, but maybe it should be
`.slice(pageNumber, pageNumber + pageSize` ?

Reply:
No, it should not

For example:
`PageNumber `= 5
`PageSize `= 20

In my code it will be `slice(100, 120)`
In your assumption it will be `slice(5, 25)`

Reply:
You're correct. It's okay then